lib/statsd.js

```diff
@@ -62,6 +62,15 @@ function makeStatsD(options, logger) {
         statsd = new StatsD(statsdOptions);
     }
 
+    // Support creating sub-metric clients with a fixed prefix. This is useful
+    // for systematically categorizing metrics per-request, by using a
+    // specific logger.
+    statsd.makeChild = function(name) {
+        var childOptions = Object.assign({}, options);
+        childOptions._prefix = statsdOptions.prefix + normalizeName(name) + '.';
```

Comment:
I don't see where this is checked. But, to be more exact, if the prefix is null / undefined / empty string, then the name of the parent and child is equal, which should be a bug, IMHO, since as it stands, one does not have to declare the prefix options in order to use `makeChild()`.

Also, the parent metrics object's name does not end with a dot, while the child's name will (as presented on line 40).